src/Bicep.Core.IntegrationTests/ExtensionRegistryTests.cs

```diff
@@ -164,35 +167,35 @@ extension myExtension
     public async Task Missing_extension_file_raises_a_diagnostic()
     {
         // See https://github.com/Azure/bicep/issues/14770 for context
-        var result = await CompilationHelper.RestoreAndCompile(
-          ("main.bicep", new("""
-extension './non_existent.tgz'
-""")));
+        var result = await new TestCompiler().RestoreAndCompileMockFileSystemFiles(
```

Comment:
Only the failing tests have been updated to use the new `TestCompiler` for now. Migrating the remaining tests will take time and will be done incrementally.